src/Polly.Core/CircuitBreaker/Controller/CircuitStateController.cs

```diff
@@ -285,9 +296,13 @@ private void CloseCircuit_NeedsLock(Outcome<T> outcome, bool manual, ResilienceC
 
             if (_onClosed is not null)
             {
-                _executor.ScheduleTask(() => _onClosed(args).AsTask(), context, out scheduledTask);
+                return _executor.ScheduleTask(() => _onClosed(args).AsTask());
             }
         }
+
+#pragma warning disable S4586
+        return null;
+#pragma warning restore S4586
```

Comment:
Rather than need to suppress things like this, could we just return `Task.CompletedTask` and do `!ReferenceEquals(task, Task.CompletedTask)` instead of `task != null` in the relevant places?